src/handlers/dns/v2/DnsStreamHandler.h

```diff
@@ -58,135 +64,209 @@ enum Protocol : uint64_t {
+        {
+            xacts.to_prometheus(out, add_labels);
+            UDP.to_prometheus(out, add_labels);
+            TCP.to_prometheus(out, add_labels);
+            DOH.to_prometheus(out, add_labels);
+            DOT.to_prometheus(out, add_labels);
+            IPv4.to_prometheus(out, add_labels);
+            IPv6.to_prometheus(out, add_labels);
+            NX.to_prometheus(out, add_labels);
+            ECS.to_prometheus(out, add_labels);
+            REFUSED.to_prometheus(out, add_labels);
+            SRVFAIL.to_prometheus(out, add_labels);
+            RNOERROR.to_prometheus(out, add_labels);
+            NODATA.to_prometheus(out, add_labels);
+            timeout.to_prometheus(out, add_labels);
+            orphan.to_prometheus(out, add_labels);
         }
     };
-    counters _counters;
+    Counters counters;
+
+    Quantile<uint64_t> dnsTimeUs;
+    Quantile<double> dnsRatio;
+
+    Cardinality qnameCard;
+
+    TopN<visor::geo::City> topGeoLocECS;
+    TopN<std::string> topASNECS;
+    TopN<std::string> topQueryECS;
+    TopN<std::string> topQname2;
+    TopN<std::string> topQname3;
+    TopN<std::string> topNX;
+    TopN<std::string> topREFUSED;
+    TopN<std::string> topSizedQnameResp;
+    TopN<std::string> topSRVFAIL;
+    TopN<std::string> topNODATA;
+    TopN<std::string> topNOERROR;
+    TopN<uint16_t> topUDPPort;
+    TopN<uint16_t> topQType;
+    TopN<uint16_t> topRCode;
+    TopN<std::string> topSlow;
+
+    DnsDirection()
+        : counters()
+        , dnsTimeUs(DNS_SCHEMA, {"xact_time_us"}, "Quantiles of transaction timing (query/reply pairs) in microseconds")
+        , dnsRatio(DNS_SCHEMA, {"response_query_size_ratio"}, "Quantiles of ratio of packet sizes in a DNS transaction (reply/query)")
+        , qnameCard(DNS_SCHEMA, {"cardinality", "qname"}, "Cardinality of unique QNAMES, both ingress and egress")
+        , topGeoLocECS(DNS_SCHEMA, "geo_loc", {"top_geo_loc_ecs_xacts"}, "Top GeoIP ECS locations")
+        , topASNECS(DNS_SCHEMA, "asn", {"top_asn_ecs_xacts"}, "Top ASNs by ECS")
+        , topQueryECS(DNS_SCHEMA, "ecs", {"top_query_ecs_xacts"}, "Top EDNS Client Subnet (ECS) observed in DNS queries")
+        , topQname2(DNS_SCHEMA, "qname", {"top_qname2_xacts"}, "Top QNAMES, aggregated at a depth of two labels")
+        , topQname3(DNS_SCHEMA, "qname", {"top_qname3_xacts"}, "Top QNAMES, aggregated at a depth of three labels")
+        , topNX(DNS_SCHEMA, "qname", {"top_nxdomain_xacts"}, "Top QNAMES with result code NXDOMAIN")
+        , topREFUSED(DNS_SCHEMA, "qname", {"top_refused_xacts"}, "Top QNAMES with result code REFUSED")
+        , topSizedQnameResp(DNS_SCHEMA, "qname", {"top_response_bytes"}, "Top QNAMES by response volume in bytes")
+        , topSRVFAIL(DNS_SCHEMA, "qname", {"top_srvfail_xacts"}, "Top QNAMES with result code SRVFAIL")
+        , topNODATA(DNS_SCHEMA, "qname", {"top_nodata_xacts"}, "Top QNAMES with result code NOERROR and no answer section")
```

Comment:
"Top QNAMES with result code NOERROR and empty answer section"

Reply:
done